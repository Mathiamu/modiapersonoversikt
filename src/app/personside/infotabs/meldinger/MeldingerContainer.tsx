import * as React from 'react';
import { Traad } from '../../../../models/meldinger/meldinger';
import styled from 'styled-components';
import theme, { pxToRem } from '../../../../styles/personOversiktTheme';
import RestResourceConsumer from '../../../../rest/consumer/RestResourceConsumer';
import TraadListe from './traadliste/TraadListe';
import { CenteredLazySpinner } from '../../../../components/LazySpinner';
import { useEffect } from 'react';
import { hasData } from '../../../../rest/utils/restResource';
import { huskForrigeValgtTraad } from '../../../../redux/meldinger/actions';
import { useDispatch } from 'react-redux';
import { useAppState, useRestResource } from '../../../../utils/customHooks';
import { erValgtIDyplenke, MeldingerDyplenkeRouteComponentProps, useInfotabsDyplenker } from '../dyplenker';
import { withRouter } from 'react-router';
import TraadVisning from './traadvisning/TraadVisning';
import Verktoylinje from './traadvisning/verktoylinje/Verktoylinje';

const meldingerMediaTreshold = pxToRem(900);

const MeldingerArticleStyle = styled.article`
    @media (min-width: ${meldingerMediaTreshold}) {
        display: flex;
        align-items: flex-start;
        > *:last-child {
            margin-left: ${theme.margin.layout};
        }
    }
    > * {
        margin-bottom: ${theme.margin.layout};
    }
    position: relative;
    > *:first-child {
        flex: 30% 1 1;
    }
    > *:last-child {
        position: sticky;
        top: 0;
        flex: 70% 1 1;
    }
`;

function MeldingerContainer(props: MeldingerDyplenkeRouteComponentProps) {
    const dispatch = useDispatch();
    const forrigeValgteTraad = useAppState(state => state.meldinger.forrigeValgteTraad);
    const traaderResource = useRestResource(resources => resources.tråderOgMeldinger);
    const dyplenker = useInfotabsDyplenker();

    useEffect(() => {
        if (!hasData(traaderResource)) {
            return;
        }
        const traadIUrl = traaderResource.data.find(traad => erValgtIDyplenke.meldinger(traad, props));
        if (!traadIUrl) {
            props.history.push(dyplenker.meldinger.link(forrigeValgteTraad || traaderResource.data[0]));
        } else if (traadIUrl !== forrigeValgteTraad) {
            dispatch(huskForrigeValgtTraad(traadIUrl));
        }
    }, [forrigeValgteTraad, traaderResource, dispatch, props, dyplenker.meldinger]);

    return (
        <RestResourceConsumer<Traad[]>
            getResource={restResources => restResources.tråderOgMeldinger}
            returnOnPending={<CenteredLazySpinner />}
        >
            {data => (
                <MeldingerArticleStyle>
                    <TraadListe traader={data} />
                    <div>
                        <Verktoylinje />
                        <TraadVisning />
                    </div>
                </MeldingerArticleStyle>
            )}
        </RestResourceConsumer>
    );
}

export default withRouter(MeldingerContainer);
