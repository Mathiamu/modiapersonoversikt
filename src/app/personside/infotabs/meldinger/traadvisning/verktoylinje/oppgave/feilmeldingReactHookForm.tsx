import * as React from 'react';
import { FieldPath, FieldValues } from 'react-hook-form';

export const feilmeldingReactHookForm = <
    TFieldValues extends FieldValues,
    TFieldName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>
>(
    { formState }: FieldValues,
    formKey: TFieldName
) => {
    const { touchedFields, isSubmitted, errors } = formState;

    const touched = touchedFields[formKey];
    const error = errors[formKey];

    if ((touched || isSubmitted) && error) {
        return <>{error.message}</>;
    }
    return undefined;
};
