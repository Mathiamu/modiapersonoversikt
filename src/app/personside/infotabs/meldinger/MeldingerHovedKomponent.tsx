import * as React from 'react';
import { Traad } from '../../../../models/meldinger/meldinger';
import TraadListeContainer from './traadliste/TraadListeContainer';

interface VisningProps {
    traader: Traad[];
}

function MeldingerVisning(props: VisningProps) {
    return <TraadListeContainer />;
}

export default MeldingerVisning;
