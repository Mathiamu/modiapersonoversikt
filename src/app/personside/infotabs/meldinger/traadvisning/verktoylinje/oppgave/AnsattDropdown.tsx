import React, { useCallback } from 'react';
import { Ansatt } from '../../../../../../../models/meldinger/oppgave';
import Downshift from 'downshift';
import theme from '../../../../../../../styles/personOversiktTheme';
import { Normaltekst } from 'nav-frontend-typografi';
import styled from 'styled-components';
import { getMockAnsatte } from '../../../../../../../mock/meldinger/oppgave-mock';
import { Input } from 'nav-frontend-skjema';

const DropDownWrapper = styled.div`
    position: relative;
    ul {
        z-index: 1000;
        position: absolute;
        box-shadow: 0 0 1rem rgba(0, 0, 0, 0.2);
    }
    li {
        &:hover,
        &:focus {
            background-color: ${theme.color.navLysGra};
        }
        min-width: 20rem;
        padding: 0.5rem 1rem;
        border: solid 0.05rem rgba(0, 0, 0, 0.2);
        background-color: white;
        color: black;
        display: flex;
    }
`;

interface Props {
    setValue: (value: Ansatt) => void;
    inputValue: Ansatt | undefined;
}

function AnsattDropdown(props: Props) {
    const handleStateChange = useCallback(
        (changes: any) => {
            if (changes.hasOwnProperty('selectedItem')) {
                props.setValue(changes.selectedItem);
            }
        },
        [props]
    );

    const suggestions: Ansatt[] = getMockAnsatte();

    return (
        <Downshift
            selectedItem={props.inputValue}
            onStateChange={handleStateChange}
            itemToString={(item: Ansatt) => (item ? item.fornavn + ' ' + item.etternavn : '')}
        >
            {helpers => (
                <div>
                    <Input
                        // @ts-ignore
                        {...helpers.getInputProps({
                            onChange: e => {
                                if (e.target.value === '') {
                                    helpers.clearSelection();
                                }
                            }
                        })}
                        label={'Velg ansatt'}
                        onFocus={() => helpers.openMenu()}
                    />
                    {helpers.isOpen ? (
                        <DropDownWrapper>
                            <ul>
                                {suggestions
                                    .filter(
                                        item =>
                                            !helpers.inputValue ||
                                            item.fornavn.toLowerCase().includes(helpers.inputValue.toLowerCase()) ||
                                            item.etternavn.toLowerCase().includes(helpers.inputValue.toLowerCase()) ||
                                            item.ident.toLowerCase().includes(helpers.inputValue.toLowerCase())
                                    )
                                    .map((item, index) => (
                                        <li
                                            {...helpers.getItemProps({
                                                key: item.ident,
                                                index,
                                                item,
                                                style: {
                                                    backgroundColor:
                                                        helpers.selectedItem === item
                                                            ? theme.color.navLysGra
                                                            : '#ffffff',
                                                    fontWeight: helpers.selectedItem === item ? 'bold' : 'normal'
                                                }
                                            })}
                                        >
                                            <Normaltekst>
                                                {item.fornavn} {item.etternavn} ({item.ident})
                                            </Normaltekst>
                                        </li>
                                    ))}
                            </ul>
                        </DropDownWrapper>
                    ) : null}
                </div>
            )}
        </Downshift>
    );
}

export default AnsattDropdown;
