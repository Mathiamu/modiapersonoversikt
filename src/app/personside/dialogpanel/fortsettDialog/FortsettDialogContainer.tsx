import * as React from 'react';
import { FormEvent, useState } from 'react';
import FortsettDialog from './FortsettDialog';
import { FortsettDialogValidator } from './validatorer';
import { ForsettDialogRequest, Meldingstype, SendDelsvarRequest, Traad } from '../../../../models/meldinger/meldinger';
import { setIngenValgtTraadDialogpanel } from '../../../../redux/oppgave/actions';
import { useFødselsnummer, useRestResource } from '../../../../utils/customHooks';
import { useDispatch } from 'react-redux';
import { OppgavelisteValg } from '../sendMelding/SendNyMelding';
import LeggTilbakepanel from './leggTilbakePanel/LeggTilbakepanel';
import {
    DelsvarRegistrertKvittering,
    OppgaveLagtTilbakeKvittering,
    SvarSendtKvittering
} from './FortsettDialogKvittering';
import useOpprettHenvendelse from './useOpprettHenvendelse';
import { erEldsteMeldingJournalfort } from '../../infotabs/meldinger/utils/meldingerUtils';
import { loggError, loggEvent } from '../../../../utils/frontendLogger';
import { post } from '../../../../api/api';
import { apiBaseUri } from '../../../../api/config';
import {
    DialogPanelStatus,
    FortsettDialogPanelState,
    FortsettDialogState,
    KvitteringsData
} from './FortsettDialogTypes';

export type FortsettDialogType =
    | Meldingstype.SVAR_SKRIFTLIG
    | Meldingstype.DELVIS_SVAR_SKRIFTLIG
    | Meldingstype.SVAR_OPPMOTE
    | Meldingstype.SVAR_TELEFON
    | Meldingstype.SPORSMAL_MODIA_UTGAAENDE;

interface Props {
    traad: Traad;
}

function FortsettDialogContainer(props: Props) {
    const initialState = {
        tekst: '',
        dialogType: Meldingstype.SVAR_SKRIFTLIG as FortsettDialogType,
        tema: undefined,
        visFeilmeldinger: false,
        sak: undefined,
        oppgaveListe: OppgavelisteValg.MinListe
    };
    const [state, setState] = useState<FortsettDialogState>(initialState);
    const reloadMeldinger = useRestResource(resources => resources.tråderOgMeldinger.actions.reload);
    const resetPlukkOppgaveResource = useRestResource(resources => resources.plukkNyeOppgaver.actions.reset);
    const reloadTildelteOppgaver = useRestResource(resources => resources.tildelteOppgaver.actions.reload);
    const fnr = useFødselsnummer();
    const [dialogStatus, setDialogStatus] = useState<FortsettDialogPanelState>({
        type: DialogPanelStatus.UNDER_ARBEID
    });
    const dispatch = useDispatch();
    const updateState = (change: Partial<FortsettDialogState>) =>
        setState({
            ...state,
            visFeilmeldinger: false,
            ...change
        });

    const opprettHenvendelse = useOpprettHenvendelse(props.traad);

    if (dialogStatus.type === DialogPanelStatus.SVAR_SENDT) {
        return <SvarSendtKvittering kvitteringsData={dialogStatus.kvitteringsData} />;
    }
    if (dialogStatus.type === DialogPanelStatus.DELSVAR_SENDT) {
        return <DelsvarRegistrertKvittering kvitteringsData={dialogStatus.kvitteringsData} />;
    }
    if (dialogStatus.type === DialogPanelStatus.OPPGAVE_LAGT_TILBAKE) {
        return <OppgaveLagtTilbakeKvittering payload={dialogStatus.payload} />;
    }

    if (opprettHenvendelse.success === false) {
        return opprettHenvendelse.placeholder;
    }
    const oppgaveId = opprettHenvendelse.henvendelse.oppgaveId;

    const handleAvbryt = () => dispatch(setIngenValgtTraadDialogpanel());

    const handleSubmit = (event: FormEvent) => {
        event.preventDefault();
        if (dialogStatus.type === DialogPanelStatus.POSTING) {
            return;
        }
        const callback = () => {
            dispatch(resetPlukkOppgaveResource);
            dispatch(reloadTildelteOppgaver);
            dispatch(reloadMeldinger);
        };

        const erOppgaveTilknyttetAnsatt = state.oppgaveListe === OppgavelisteValg.MinListe;
        const commonPayload = {
            fritekst: state.tekst,
            meldingstype: state.dialogType,
            traadId: props.traad.traadId,
            behandlingsId: opprettHenvendelse.henvendelse.behandlingsId,
            oppgaveId: oppgaveId
        };
        if (
            FortsettDialogValidator.erGyldigSvarSkriftlig(state) ||
            FortsettDialogValidator.erGyldigSvarOppmote(state) ||
            FortsettDialogValidator.erGyldigSvarTelefon(state)
        ) {
            setDialogStatus({ type: DialogPanelStatus.POSTING });
            const request: ForsettDialogRequest = {
                ...commonPayload,
                erOppgaveTilknyttetAnsatt: true // TODO, denne bør ikke være nødvendig å sende med her
            };
            const kvitteringsData = {
                fritekst: request.fritekst,
                meldingstype: request.meldingstype
            };
            post(`${apiBaseUri}/dialog/${fnr}/fortsett/ferdigstill`, request)
                .then(() => {
                    callback();
                    setDialogStatus({ type: DialogPanelStatus.SVAR_SENDT, kvitteringsData: kvitteringsData });
                    loggEvent('Send-Svar', 'FortsettDialog');
                })
                .catch(() => {
                    setDialogStatus({ type: DialogPanelStatus.ERROR });
                    loggEvent('Send-Svar-Failed', 'FortsettDialog');
                });
        } else if (FortsettDialogValidator.erGyldigSpørsmålSkriftlig(state, props.traad)) {
            const erJournalfort = erEldsteMeldingJournalfort(props.traad);
            if (!state.sak && !erJournalfort) {
                const error = Error('For å opprette spørsmål må meldingen være journalført eller sak må være valgt');
                console.error(error);
                loggError(error);
                return;
            }
            setDialogStatus({ type: DialogPanelStatus.POSTING });
            const request: ForsettDialogRequest = {
                ...commonPayload,
                erOppgaveTilknyttetAnsatt: erOppgaveTilknyttetAnsatt,
                sak: state.sak ? state.sak : undefined
            };
            const kvitteringsData = {
                fritekst: request.fritekst,
                meldingstype: request.meldingstype
            };
            post(`${apiBaseUri}/dialog/${fnr}/fortsett/ferdigstill`, request)
                .then(() => {
                    callback();
                    setDialogStatus({ type: DialogPanelStatus.SVAR_SENDT, kvitteringsData: kvitteringsData });
                    loggEvent('Send-Spørsmål', 'FortsettDialog');
                })
                .catch(() => {
                    setDialogStatus({ type: DialogPanelStatus.ERROR });
                    loggEvent('Send-Spørsmål-Failed', 'FortsettDialog');
                });
        } else if (FortsettDialogValidator.erGyldigDelsvar(state) && oppgaveId && state.temagruppe) {
            setDialogStatus({ type: DialogPanelStatus.POSTING });
            const request: SendDelsvarRequest = {
                fritekst: state.tekst,
                traadId: props.traad.traadId,
                oppgaveId: oppgaveId,
                temagruppe: state.temagruppe,
                behandlingsId: opprettHenvendelse.henvendelse.behandlingsId
            };
            post(`${apiBaseUri}/dialog/${fnr}/delvis-svar`, request)
                .then(() => {
                    callback();
                    const kvitteringsData: KvitteringsData = {
                        fritekst: request.fritekst,
                        meldingstype: Meldingstype.DELVIS_SVAR_SKRIFTLIG,
                        temagruppe: request.temagruppe
                    };
                    setDialogStatus({ type: DialogPanelStatus.DELSVAR_SENDT, kvitteringsData: kvitteringsData });
                    loggEvent('Send-Delsvar', 'FortsettDialog');
                })
                .catch(() => {
                    setDialogStatus({ type: DialogPanelStatus.ERROR });
                    loggEvent('Send-Delsvar-Failed', 'FortsettDialog');
                });
        } else {
            updateState({ visFeilmeldinger: true });
        }
    };

    const meldingMedTemagruppe = props.traad.meldinger.find(melding => melding.temagruppe);
    const temagruppe = meldingMedTemagruppe ? meldingMedTemagruppe.temagruppe : undefined;

    return (
        <>
            <FortsettDialog
                handleAvbryt={handleAvbryt}
                state={state}
                updateState={updateState}
                handleSubmit={handleSubmit}
                traad={props.traad}
                key={props.traad.traadId}
                fortsettDialogPanelState={dialogStatus}
                erTilknyttetOppgave={!!oppgaveId}
            />
            {oppgaveId && (
                <LeggTilbakepanel
                    oppgaveId={oppgaveId}
                    status={dialogStatus}
                    setDialogStatus={setDialogStatus}
                    temagruppe={temagruppe}
                />
            )}
        </>
    );
}

export default FortsettDialogContainer;
