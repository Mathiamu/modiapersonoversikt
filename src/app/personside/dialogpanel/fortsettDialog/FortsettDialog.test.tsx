import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TestProvider from '../../../../test/Testprovider';
import { getTestStore } from '../../../../test/testStore';
import { setDialogpanelTraad } from '../../../../redux/oppgave/actions';
import { statiskTraadMock } from '../../../../mock/meldinger/statiskTraadMock';
import FortsettDialogContainer from './FortsettDialogContainer';

test('viser fortsett dialog', () => {
    const testStore = getTestStore();
    testStore.dispatch(setDialogpanelTraad(statiskTraadMock));
    const dialogPanelBody = renderer.create(
        <TestProvider customStore={testStore}>
            <FortsettDialogContainer traad={statiskTraadMock} />
        </TestProvider>
    );

    expect(dialogPanelBody.toJSON()).toMatchSnapshot();
});
