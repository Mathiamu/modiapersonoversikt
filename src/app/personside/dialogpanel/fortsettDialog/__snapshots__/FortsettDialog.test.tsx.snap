// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viser fortsett dialog 1`] = `
.c3 .snakkeboble-panel {
  -webkit-flex-basis: 70%;
  -ms-flex-preferred-size: 70%;
  flex-basis: 70%;
}

.c3 .nav-ikon,
.c3 .bruker-ikon {
  background-size: contain;
  background-repeat: no-repeat;
  background-position-y: 0.5rem;
}

.c3 .nav-ikon {
  background-image: url(navlogo.svg);
  width: 3.5rem;
}

.c3 .bruker-ikon {
  background-image: url(bruker.svg);
  width: 2.5rem;
}

.c2 {
  margin-bottom: 1rem;
}

.c2 .ekspanderbartPanel__hode {
  padding: 0.6rem;
}

.c5 label {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px 1px 1px 1px);
  clip: rect(1px 1px 1px 1px);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c4 textarea {
  min-height: 9rem;
}

.c4 label {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px 1px 1px 1px);
  clip: rect(1px 1px 1px 1px);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c7 label {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px 1px 1px 1px);
  clip: rect(1px 1px 1px 1px);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c9 .ekspanderbartPanel__hode {
  padding: 0.6rem;
}

.c11 > *:not(:first-child) {
  margin-top: 0.3rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c10 > * {
  margin-top: 0.5rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 1rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c1 .skjemaelement {
  margin-bottom: 0;
}

.c1 > * {
  margin-bottom: 1rem;
}

.c6 > *:not(:first-child) {
  margin-top: 1rem;
}

.c0 {
  padding: 1rem .8rem;
}

.c8 {
  margin-top: 1rem;
}

<article
  className="c0"
>
  <h2
    className="typo-undertittel"
  >
    Fortsett dialog
  </h2>
  <form
    className="c1"
    onSubmit={[Function]}
  >
    <div
      className="ekspanderbartPanel c2 ekspanderbartPanel--lukket ekspanderbartPanel--border"
    >
      <button
        aria-expanded={false}
        className="ekspanderbartPanel__hode"
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <div
          className="ekspanderbartPanel__flex-wrapper"
        >
          <span
            className="typo-normal ekspanderbartPanel__heading"
          >
            Tidligere meldinger
          </span>
          <span
            className="ekspanderbartPanel__indikator"
          />
        </div>
      </button>
      <article
        aria-label="Tidligere meldinger"
        className="ekspanderbartPanel__innhold"
      >
        <div
          className="c3"
        >
          <div
            className="snakkeboble snakkeboble--hoyre"
          >
            <i
              className="nav-ikon"
            />
            <div
              className="snakkeboble__snakkeboble-pil-container"
            >
              <i
                className="snakkeboble__snakkebole-pil"
              />
            </div>
            <div
              className="panel snakkeboble-panel"
            >
              <div
                className="snakkeboble-panel__tekst"
              >
                <p
                  className="typo-element"
                >
                  Samtalereferat oppmøte - Lest,  Arbeid
                </p>
                <p
                  className="typo-normal"
                >
                  30.07.2019 00:00
                </p>
                <p
                  className="typo-normal"
                >
                  Skrevet av: 
                  Sander Hansen (nzxo4q)
                </p>
                <hr />
                <p
                  className="typo-normal"
                >
                  Ea laborum accusamus et ex voluptates provident eveniet voluptatem. Sunt et sed eligendi harum exercitationem. Eaque est maiores autem sunt occaecati atque ipsa ea itaque. Dolorum enim quaerat alias ut sit labore pariatur.
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="c3"
        >
          <div
            className="snakkeboble snakkeboble--venstre"
          >
            <i
              className="bruker-ikon"
            />
            <div
              className="snakkeboble__snakkeboble-pil-container"
            >
              <i
                className="snakkeboble__snakkebole-pil"
              />
            </div>
            <div
              className="panel snakkeboble-panel"
            >
              <div
                className="snakkeboble-panel__tekst"
              >
                <p
                  className="typo-element"
                >
                  Spørsmål fra bruker - Ulest,  Arbeid
                </p>
                <p
                  className="typo-normal"
                >
                  10.08.2019 00:00
                </p>
                <p
                  className="typo-normal"
                >
                  Skrevet av: 
                  Frida Pedersen (xmqgb8)
                </p>
                <hr />
                <p
                  className="typo-normal"
                >
                  Quo magni eaque et libero minima ipsum amet et dolorem. Sed iusto ratione sit. Voluptate alias qui impedit ut dolorum doloribus qui. Qui beatae rerum minima suscipit.
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="c3"
        >
          <div
            className="snakkeboble snakkeboble--venstre"
          >
            <i
              className="bruker-ikon"
            />
            <div
              className="snakkeboble__snakkeboble-pil-container"
            >
              <i
                className="snakkeboble__snakkebole-pil"
              />
            </div>
            <div
              className="panel snakkeboble-panel"
            >
              <div
                className="snakkeboble-panel__tekst"
              >
                <p
                  className="typo-element"
                >
                  Spørsmål fra bruker - Lest,  Arbeid
                </p>
                <p
                  className="typo-normal"
                >
                  06.08.2019 00:00
                </p>
                <p
                  className="typo-normal"
                >
                  Skrevet av: 
                  Synne Lunde (e5otkk)
                </p>
                <hr />
                <p
                  className="typo-normal"
                >
                  Animi qui molestiae omnis reiciendis quam in. Provident magnam dolor. Omnis voluptatem placeat labore in ut aliquam perspiciatis fugiat. Perspiciatis voluptatum odit.
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="c3"
        >
          <div
            className="snakkeboble snakkeboble--hoyre"
          >
            <i
              className="nav-ikon"
            />
            <div
              className="snakkeboble__snakkeboble-pil-container"
            >
              <i
                className="snakkeboble__snakkebole-pil"
              />
            </div>
            <div
              className="panel snakkeboble-panel"
            >
              <div
                className="snakkeboble-panel__tekst"
              >
                <p
                  className="typo-element"
                >
                  Delsvar - Lest,  Arbeid
                </p>
                <p
                  className="typo-normal"
                >
                  13.08.2019 00:00
                </p>
                <p
                  className="typo-normal"
                >
                  Skrevet av: 
                  Kristoffer Fredriksen (xy70r8)
                </p>
                <hr />
                <p
                  className="typo-normal"
                >
                  Dolores omnis ea voluptatem corrupti odit incidunt enim. Unde fugiat sed animi explicabo maiores. Nihil cum accusamus omnis officiis deserunt est. Qui quisquam unde aut magni dignissimos.
                </p>
              </div>
            </div>
          </div>
        </div>
      </article>
    </div>
    <div
      className="c4"
    >
      <div
        className="skjemaelement textarea__container"
      >
        <label
          className="skjemaelement__label"
          htmlFor="Helt tilfeldig ID"
        >
          Melding
        </label>
        <div
          className="textarea--medMeta__wrapper"
        >
          <textarea
            className="skjemaelement__input textarea--medMeta"
            id="Helt tilfeldig ID"
            onChange={[Function]}
            placeholder="Alt du skriver i denne boksen blir synlig for Aremark når du trykker \\"Del med Aremark\\""
            style={
              Object {
                "height": "30px",
              }
            }
            type="text"
            value=""
          />
          <p
            className="textarea--medMeta__teller"
          >
            <span>
              Du har 
              5000
               tegn igjen
            </span>
          </p>
        </div>
        <div
          aria-live="assertive"
          role="alert"
        />
        <div
          aria-hidden="true"
          className="textareamirror"
        />
      </div>
      <div
        aria-live="assertive"
        role="alert"
      />
    </div>
    <div>
      <div
        className="skjemaelement c5"
      >
        <label
          className="skjemaelement__label"
          htmlFor="Helt tilfeldig ID"
        >
          Velg dialogtype
        </label>
        <div
          className="selectContainer input--fullbredde"
        >
          <select
            className="skjemaelement__input"
            disabled={false}
            id="Helt tilfeldig ID"
            onChange={[Function]}
            value="SVAR_SKRIFTLIG"
          >
            <option
              value="SVAR_SKRIFTLIG"
            >
              Svar skriftlig
            </option>
            <option
              value="DELVIS_SVAR_SKRIFTLIG"
            >
              Svar delvis
            </option>
            <option
              value="SVAR_TELEFON"
            >
              Svar skriftlig telefon
            </option>
            <option
              value="SVAR_OPPMOTE"
            >
              Svar skriftlig oppmøte
            </option>
          </select>
        </div>
        <div
          aria-live="assertive"
          role="alert"
        />
      </div>
    </div>
    <div
      className=""
    >
      <div
        className="alertstripe alertstripe--info"
      >
        <span
          aria-label="info"
          className="alertstripe__ikon"
        >
          <svg
            focusable="false"
            height="1.5em"
            kind="info-sirkel-fyll"
            viewBox="0 0 24 24"
            width="1.5em"
          >
            <g
              fill="none"
              fillRule="nonzero"
            >
              <path
                d="M12 0C5.382 0 0 5.382 0 12s5.382 12 12 12c6.617 0 12-5.382 12-12S18.617 0 12 0z"
                fill="#669DB4"
              />
              <g
                fill="#FFF"
              >
                <path
                  d="M12 5a1.566 1.566 0 1 1 .11 3.13A1.566 1.566 0 0 1 12 5zM14.976 17.01c.563 0 1.043.431 1.043.991s-.48.992-1.043.992H9.39c-.564 0-1.043-.431-1.043-.992 0-.56.479-.99 1.043-.99h1.6v-5.016h-.986c-.565 0-1.044-.43-1.044-.991 0-.56.48-.991 1.044-.991h2.03c.563 0 1.043.43 1.043.99v6.007H14.976z"
                />
              </g>
            </g>
          </svg>
        </span>
        <div
          className="typo-normal alertstripe__tekst"
        >
          Bruker kan ikke svare
        </div>
      </div>
      <fieldset
        className="c6"
      >
        <div
          className="skjemaelement skjemaelement--horisontal"
        >
          <input
            checked={false}
            className="skjemaelement__input checkboks"
            id="Helt tilfeldig ID"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="skjemaelement__label"
            htmlFor="Helt tilfeldig ID"
          >
            Bruker kan svare
          </label>
          <div
            aria-live="assertive"
            role="alert"
          />
        </div>
      </fieldset>
      <div
        className=""
      >
        <div
          className="skjemaelement c7"
        >
          <label
            className="skjemaelement__label"
            htmlFor="Helt tilfeldig ID"
          >
            Tema
          </label>
          <div
            className="selectContainer input--fullbredde"
          >
            <select
              className="skjemaelement__input"
              disabled={false}
              id="Helt tilfeldig ID"
              onChange={[Function]}
              value=""
            >
              <option
                disabled={true}
                value=""
              >
                Velg temagruppe
              </option>
              <option
                value="ARBD"
              >
                Arbeid
              </option>
              <option
                value="FMLI"
              >
                Familie
              </option>
              <option
                value="HJLPM"
              >
                Hjelpemiddel
              </option>
              <option
                value="PENS"
              >
                Pensjon
              </option>
              <option
                value="OVRG"
              >
                Øvrig
              </option>
            </select>
          </div>
          <div
            aria-live="assertive"
            role="alert"
          />
        </div>
        <div
          aria-live="assertive"
          role="alert"
        />
      </div>
    </div>
    <button
      className="knapp c8 knapp--hoved"
      disabled={false}
      type="submit"
    >
      Del med Aremark
    </button>
    <div
      className="ekspanderbartPanel c9 ekspanderbartPanel--lukket ekspanderbartPanel--border"
    >
      <button
        aria-expanded={false}
        className="ekspanderbartPanel__hode"
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <div
          className="ekspanderbartPanel__flex-wrapper"
        >
          <span
            className="typo-normal ekspanderbartPanel__heading"
          >
            Legg tilbake
          </span>
          <span
            className="ekspanderbartPanel__indikator"
          />
        </div>
      </button>
      <article
        aria-label="Legg tilbake"
        className="ekspanderbartPanel__innhold"
      >
        <div
          className="c10"
        >
          <fieldset
            className="c11"
          >
            <legend
              className="sr-only"
            >
              Velg årsak
            </legend>
            <div
              className="skjemaelement skjemaelement--horisontal"
            >
              <input
                checked={false}
                className="skjemaelement__input radioknapp"
                id="Helt tilfeldig ID"
                name="årsak"
                onChange={[Function]}
                type="radio"
              />
              <label
                className="skjemaelement__label"
                htmlFor="Helt tilfeldig ID"
              >
                Innhabil
              </label>
            </div>
            <div
              className="skjemaelement skjemaelement--horisontal"
            >
              <input
                checked={false}
                className="skjemaelement__input radioknapp"
                id="Helt tilfeldig ID"
                name="årsak"
                onChange={[Function]}
                type="radio"
              />
              <label
                className="skjemaelement__label"
                htmlFor="Helt tilfeldig ID"
              >
                Feil temagruppe
              </label>
            </div>
            <div
              className=""
            >
              <div
                className="skjemaelement c7"
              >
                <label
                  className="skjemaelement__label"
                  htmlFor="Helt tilfeldig ID"
                >
                  Tema
                </label>
                <div
                  className="selectContainer input--fullbredde"
                >
                  <select
                    className="skjemaelement__input"
                    disabled={false}
                    id="Helt tilfeldig ID"
                    onChange={[Function]}
                    value=""
                  >
                    <option
                      disabled={true}
                      value=""
                    >
                      Velg temagruppe
                    </option>
                    <option
                      value="ARBD"
                    >
                      Arbeid
                    </option>
                    <option
                      value="FMLI"
                    >
                      Familie
                    </option>
                    <option
                      value="HJLPM"
                    >
                      Hjelpemiddel
                    </option>
                    <option
                      value="PENS"
                    >
                      Pensjon
                    </option>
                    <option
                      value="OVRG"
                    >
                      Øvrig
                    </option>
                  </select>
                </div>
                <div
                  aria-live="assertive"
                  role="alert"
                />
              </div>
              <div
                aria-live="assertive"
                role="alert"
              />
            </div>
            <div
              className="skjemaelement skjemaelement--horisontal"
            >
              <input
                checked={false}
                className="skjemaelement__input radioknapp"
                id="Helt tilfeldig ID"
                name="årsak"
                onChange={[Function]}
                type="radio"
              />
              <label
                className="skjemaelement__label"
                htmlFor="Helt tilfeldig ID"
              >
                Annen årsak
              </label>
            </div>
            <div
              className="skjemaelement textarea__container"
            >
              <label
                className="skjemaelement__label"
                htmlFor="Helt tilfeldig ID"
              >
                Årsak
              </label>
              <div
                className="textarea--medMeta__wrapper"
              >
                <textarea
                  className="skjemaelement__input textarea--medMeta"
                  id="Helt tilfeldig ID"
                  onChange={[Function]}
                  style={
                    Object {
                      "height": "30px",
                    }
                  }
                  type="text"
                  value=""
                />
                <p
                  className="textarea--medMeta__teller"
                >
                  <span>
                    Du har 
                    2000
                     tegn igjen
                  </span>
                </p>
              </div>
              <div
                aria-live="assertive"
                role="alert"
              />
              <div
                aria-hidden="true"
                className="textareamirror"
              />
            </div>
          </fieldset>
          <button
            className="knapp knapp--hoved"
            disabled={false}
            type="submit"
          >
            Legg tilbake
          </button>
        </div>
      </article>
    </div>
  </form>
</article>
`;
